test/schemas/testpilot.js

```diff
@@ -44,154 +55,43 @@ describe("The 'testpilot' Schema", function() {
-  });
-  it("should reject if required field 'timestamp' is missing", function(done) {
-    const ping = clone(samplePing);
-    delete ping.payload.timestamp;
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required field 'timestamp' is a Date object", function(done) {
-    const ping = clone(samplePing);
-    ping.payload.timestamp = new Date();
-    ping.payload.events[0].timestamp = new Date().toISOString();
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required field 'timestamp' is a Date ISO string", function(done) {
-    const ping = clone(samplePing);
-    ping.payload.timestamp = new Date().toISOString();
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required field 'version' is missing", function(done) {
-    const ping = clone(samplePing);
-    delete ping.payload.version;
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required field 'version' is malformed", function(done) {
-    const ping = clone(samplePing);
-    ping.payload.version = 12345; // version is an AMO version string, not an integer
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required event field 'event' is missing", function(done) {
-    const ping = clone(samplePing);
-    delete ping.payload.events[0].event;
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required event field 'event' is malformed", function(done) {
-    const ping = clone(samplePing);
-    ping.payload.events[0].event = 1;
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
-    });
-  });
-  it("should reject if required event field 'object' is missing", function(done) {
+  it("should reject if required field topic is incorrect", function(done) {
     const ping = clone(samplePing);
-    delete ping.payload.events[0].object;
+    ping.topic = "foo";
     Joi.validate(ping, testPilotSchema, function(err) {
       expect(err).to.exist;
       done();
     });
   });
-  it("should reject if required event field 'object' is malformed", function(done) {
-    const ping = clone(samplePing);
-    ping.payload.events[0].object = 1;
-    Joi.validate(ping, testPilotSchema, function(err) {
-      expect(err).to.exist;
-      done();
+  requiredStrings.forEach((field) => {
+    it(`should reject if required field ${field} is missing`, function(done) {
+      const ping = clone(samplePing);
+      delete ping[field];
+      Joi.validate(ping, testPilotSchema, function(err) {
+        expect(err).to.exist;
```

Comment:
For giggles, I tried adding the following test, since it looks like we're [blindly trying to import files based on a user supplied topic](https://github.com/mozilla/ping-centre/blob/1a1c191fd9465c7909e9db637cfe3ca726a606b2/ping-centre.js#L18) (which now that my chubby fingers type that out, sounds mildly risky if I tried passing a topic of "../../../blah").

```js
  it("throws if cannot find schema module", function() {
    try {
      new PingCentre("bananas", "clientID");
    } catch (err) {
      assert.equal(err.message, "Cannot find module './schemas/bananas'");
    }
  });
```